app/policies/chat_channel_policy.rb

```diff
@@ -39,6 +39,18 @@ def update_channel?
     user_can_edit_channel
   end
 
+  def join_channel_invitation?
+    record.present? && user.id && record.discoverable
```

Comment:
I haven't tested it locally yet, but just by removing the discoverable check could make that last suggestion work 🙂

Reply:
yes